lib/cog/chat/slack/template_processor.ex

```diff
@@ -1,8 +1,16 @@
 defmodule Cog.Chat.Slack.TemplateProcessor do
   require Logger
 
-  def render(directives),
-    do: Enum.map_join(directives, &process_directive/1)
+  alias Cog.Util.Colors
+
+  def render(directives) do
+    {text, attachments} = directives
+                          |> Enum.map(&process_directive/1)
+                          |> List.flatten
+                          |> consolidate_outputs({"", []})
+  {text, Enum.map(attachments, fn(attachment) -> updated = Map.delete(attachment, "name")
```

Comment:
Perhaps some comments describing why we do this bit of munging, with pointers to the relevant Slack API docs?